### PR TITLE
Migrate to 1ES Pipeline for Cache Searching

### PR DESCRIPTION
**Pull request by nagilson**
Per company policy, we must migrate to this new pipeline format soon.
For further information from reviewers, please contact me for an internal link to this migration effort.

Also the log is showing changes from https://github.com/dotnet/templating/pull/7724 which was needed to get things working. 

Removes tests for resources which no longer exist and are out of support.

### CODE REVIEW

**Comment by nagilson** [comment]
Updated the SAS tokens and it got further but there appear to be more azure issues. Anyway it looks like the 1ES part of this is done, this pipeline has not worked since september so I dont want to block the migration effort by trying to fix the azure resources that were already broken before this change. This improves the pipeline by making the tests pass which did not pass before this PR.

**Comment by nagilson** [comment]
image
Dale from DDFUN was able to resolve the remaining issues on the azure PME and SAW. Pipeline is working again 🔥 
https://dev.azure.com/dnceng/internal/_build/results?buildId=2410140&view=results

File: search-cache-pipeline.yml [hunk 1]
```diff
@@ -15,160 +18,166 @@ parameters:
...
+      sourceAnalysisPool:
+        name: NetCore1ESPool-Internal
+        image: 1es-windows-2022
+        os: windows
+    pool:
+      name: NetCore1ESPool-Internal
+      image: 1es-ubuntu-2204
+      os: linux
+    customBuildTags:
+    - ES365AIMigrationTooling
```
Comment (MiYanni) [inline search-cache-pipeline.yml:41]: We don't need this. It is tracking information for using the conversion tool, which we didn't use here.

**Review by MiYanni — APPROVED** [review]

File: search-cache-pipeline.yml (hunk 1 above)
Comment (nagilson) [inline search-cache-pipeline.yml:41]:
```suggestion
```

**Pull request merged**